connection_manager_test.go

```diff
@@ -18,6 +18,20 @@ import (
 
 var vpnIp iputil.VpnIp
 
+func newTestLighthouse() *LightHouse {
+	lh := &LightHouse{
+		l:       test.NewLogger(),
+		addrMap: map[iputil.VpnIp]*RemoteList{},
+	}
+	lighthouses := map[iputil.VpnIp]struct{}{}
+	staticList := map[iputil.VpnIp]struct{}{}
+
+	lh.lighthouses.Store(&lighthouses)
+	lh.staticList.Store(&staticList)
+
+	return lh
```

Comment:
I'm not sure if this will work since `atomic.Pointer` has `noCopy` set in the struct (so no copies are allowed, which I think the return from your method would be) https://cs.opensource.google/go/go/+/refs/tags/go1.19:src/sync/atomic/type.go;l=40

Reply:
I typed up a local program to use it, and demonstrated to myself (on my M1) that it compiled and executed OK.
I saw that nocopy in the impl, and I dared the compiler to stop me...I posted the note when the compiler failed to prevent my local code snippet, figuring that I wasn't violating anything with that nocopy.

What's going on here? What does that nocopy really mean?
https://cs.opensource.google/go/go/+/refs/tags/go1.19:src/sync/atomic/type.go;l=178;drc=0f6ee42fe063a48d7825bc03097bbb714aafdb7d
leads to a github comment that explains a little more...apparently, it means very little. `go vet` will complain about a copied nocopy value, but nothing else happens. I ran this against my little local test program, and I see `go vet` complaining. So that part at least works.

But why is it wrong to copy an atomic value in the first place? Looks like it should be totally copyable to me. As long as the value is 'moved' (which admittedly doesn't exist as an enforced concept in go) it'd be just fine. Meaning, if there aren't two copies of the value in the world, I can't imagine what could possibly go wrong. Other scenarios I can think of are some underlying atomic implementation is locks under the covers, and locks are also nocopy...which would complicate this analysis. I didn't go searching to see if this was the case.

One answer that fits within this model is to change the atomics to *atomics, but that looks...suboptimal to me. I think there's not enough value in the syntactic sugar of the New method to justify yet another pointer indirection - a pointer to an atomic pointer to an ultimate value.

In conclusion, I can't find anything wrong with the code, but since the suggestion is only syntactic sugar and breaks some `go vet` output (and there may be something I'm missing! atomic code gets tricky very quickly), please do reject the helper function :D